File: uporabniki/api/health.py
# uporabniki/api/health.py
# VZOREC: Vmesnik za preverjanje stanja (Health Check)
# Vrne stanje storitve in odvisnosti (baza podatkov).

from fastapi import APIRouter
from sqlalchemy import text
from uporabniki.infrastruktura.database import SessionLocal

router = APIRouter()

@router.get("/health")
def health_check():
    """
    Preveri stanje storitve uporabnikov.
    - Preveri povezavo z bazo podatkov
    - Vrne UP/DOWN za vsako odvisnost
    """
    db_status = "UP"
    try:
        db = SessionLocal()
        db.execute(text("SELECT 1"))
        db.close()
    except Exception as e:
        db_status = f"DOWN: {str(e)}"

    overall = "UP" if db_status == "UP" else "DEGRADED"

    return {
        "status": overall,
        "service": "uporabniki",
        "dependencies": {
            "database": db_status,
        }
    }

File: projekti/src/api/health.js
// projekti/src/api/health.js
// VZOREC: Vmesnik za preverjanje stanja (Health Check)

const router = require('express').Router();
const { pool } = require('../infrastruktura/database');
const axios = require('axios');

router.get('/health', async (req, res) => {
  const checks = await Promise.all([
    checkDatabase(),
    checkNaloge(),
  ]);

  const dbCheck = checks[0];
  const nalogeCheck = checks[1];
  const allUp = checks.every(c => c.status === 'UP');

  res.status(allUp ? 200 : 207).json({
    status: allUp ? 'UP' : 'DEGRADED',
    service: 'projekti',
    dependencies: {
      database: dbCheck.status,
      naloge: nalogeCheck.status,
    },
  });
});

async function checkDatabase() {
  try {
    await pool.query('SELECT 1');
    return { status: 'UP' };
  } catch (e) {
    return { status: `DOWN: ${e.message}` };
  }
}

async function checkNaloge() {
  try {
    const url = process.env.NALOGE_URL || 'http://localhost:5001';
    await axios.get(`${url}/health`, { timeout: 2000 });
    return { status: 'UP' };
  } catch (e) {
    return { status: `DOWN: ${e.message}` };
  }
}

module.exports = router;
